lib/db/redis.js

```diff
@@ -0,0 +1,64 @@
+var redis = require('redis')
+var jws = require('jws')
+
+function match(filter, str) {
+  try { // too lazy
+    var event = JSON.parse(jws.decode(str).payload)
+    var filterNames = Object.keys(filter)
+    for (var i = 0; i < filterNames.length; i++) {
+      var name = filterNames[i]
+      if (event[name] && event[name] === filter[name]) {
+        continue
+      }
+      else {
+        return false
+      }
+    }
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+function RedisDB(options) {
+  this.db = redis.createClient(options.port, options.host)
+}
+
+RedisDB.create = function (options, cb) {
+  var x = new RedisDB(options)
+  x.db.once('ready', cb.bind(null, null, x))
+}
+
+RedisDB.prototype.append = function (str, cb) {
+  this.db.rpush('fxa', str, cb)
+}
+
+RedisDB.prototype.read = function (pos, num, filter, cb) {
+  pos = +pos || 0
+  num = num || 1000
+  filter = filter || {}
+  var matchFilter = match.bind(null, filter)
+  this.db.lrange('fxa', pos, (pos + num), function (err, results) {
```

Comment:
IIUC this will return a total of `num + 1` results, since `LRANGE` start and stop indices are inclusive.

Also the docs tell me "Time complexity: O(S+N) where S is the distance of start offset from HEAD for small lists, from nearest end (HEAD or TAIL) for large lists; and N is the number of elements in the specified range".  Which is probably OK as long as you're near the head of the list, but could be interesting if you're trying to read events from the middle.